src/js/components/form-elements/v2/DateField.jsx

```diff
@@ -26,6 +26,7 @@ const DateField = ({
   value,
   onChange,
   showTimeSelect,
+  customDateFormat,
```

Comment:
I am pretty sure, that the DatePicker already works with custom date format, so it looks unnecessary.